File: leetcode/62.UniquePaths.go
package leetcode

func uniquePaths(m int, n int) int {
	row := make([]int, n)
	for i := 0; i < n; i++ {
		row[i] = 1
	}

	for i := 0; i < m-1; i++ {
		newRow := make([]int, n)
		for j := 0; j < n; j++ {
			newRow[j] = 1
		}
		for j := n - 2; j >= 0; j-- {
			newRow[j] = newRow[j+1] + row[j]
		}
		row = newRow
	}

	return row[0]
}

File: leetcode/62.UniquePaths.ts
function uniquePaths(m: number, n: number): number {
  let row = [...new Array(n)].fill(1)

  for (let i = 0; i < m - 1; i++) {
    const newRow = [...new Array(n)].fill(1)
    for (let j = n - 2; j >= 0; j--) {
      newRow[j] = newRow[j + 1] + row[j]
    }
    row = newRow
  }

  return row[0]
}
console.log(uniquePaths(3, 7))
